### ISSUE TITLE
VSWhere stopped detecting visual studio

### ISSUE THREAD

**Issue by deadlydog** [Microsoft/vswhere#145]
We have several build servers that are all identical and configured using PowerShell DSC. Today for some reason when using the command:
```
vswhere.exe -latest -requires "Microsoft.Component.MSBuild" -property installationPath
```
on one of the build servers, it no longer returns the installation path; it does not return anything. Using just:
```
vswhere.exe -latest -requires "Microsoft.Component.MSBuild"
```
also does not return anything.

This was working fine previously, and no updates were installed on the build servers recently. We were using an older version of vswhere.exe (v2.1.3), but still have the same problem when using the latest version (v2.4.1).

![image](https://user-images.githubusercontent.com/1187140/38573019-097802b2-3cb2-11e8-921e-5d61c640bb76.png)

The same commands return the expected results on our other build servers.

Visual Studio 2017 is installed on the affected server, and I'm able to open it up without any issues. Here's the VS version info from the build server having the issue:
![image](https://user-images.githubusercontent.com/1187140/38573178-7aa28638-3cb2-11e8-9d69-24509e6edc01.png)

And it is installed at the standard default location:
![image](https://user-images.githubusercontent.com/1187140/38573261-b9ed942c-3cb2-11e8-8843-6c6c02ccf6d9.png)

Do you have any idea why VSWhere stopped working on this machine? Or suggestions for us to try to fix it?

Thanks!

**Comment by heaths** [comment]
It's possible there was an installation error or something. If you pass `-all` do you see it? The query API is registered based on the version found in the logo. You could get more information about why an instance isn't showing if you install https://github.com/Microsoft/vssetup.powershell and do:
```powershell
get-vssetupinstance -all -pre | select InstanceId, State
```

**Comment by deadlydog** [comment]
Hmmm, this is very odd. I just checked again today and now the command is working properly on that server. My guess is that PowerShell DSC must be messing with the installation somehow, as I know we are using it to install VS and keep it up to date. I'm going to close this issue, but will report back if it happens again and I can determine that DSC isn't at fault. Thanks!

![image](https://user-images.githubusercontent.com/1187140/38648347-fa0b0548-3dad-11e8-970e-c0c1c117758f.png)

**Comment by heaths** [comment]
If the DSC resource isn't itself waiting for the process to finish and if it doesn't pass `--wait`, the install is asynchronous to the DSC resource, meaning when you are checking it the install may still be in progress. Based on your description, that may be the case (without knowing the command line you used). When scripting the install, you need to pass `--wait` to the bootstrapper which tells it to wait on the actual installer.

**Comment by deadlydog** [comment]
Thanks for the quick response @heaths. That would make sense, except we were seeing this issue for 12+ hours, and I wouldn't expect a VS install to take that long. Also, I was still able to open up VS without any problems while the issue was occurring, so I think that would indicate that there wasn't actually a VS installation in progress while we were experiencing the problem; I imagine I would have received some sort of error if I tried to open Visual Studio was it was updating/installing.

**Comment by heaths** [comment]
Did the server have to restart at some point? We don't report an instance is launchable if a reboot is required for that instance. Ignoring reboots has historically (for any app, really) caused problems that are difficult to diagnose and can manifest is completely unexpected and unsupported ways.

**Comment by deadlydog** [comment]
I hadn't noticed if the server needed a reboot, but that could explain what I was seeing. I haven't seen the problem again yet, but I'll keep that in mind in case I do. Thanks.

**Comment by tellarin** [comment]
I'm having the same issue now (started today after months working fine). Rebooting the server made no difference. VS 15.9.28307.812.

**Comment by heaths** [comment]
@tellarin, please run https://aka.ms/vscollect and attach %TEMP%\vslogs.zip to a new issue on https://developercommunity.visualstudio.com.

**Comment by tellarin** [comment]
@heaths I've completely re-installed the build agent where this was happening and it doesn't happen anymore. I'll follow your suggestion if it happens again.
Could this suggested steps be added in the FAQ in case other people face the same problem in the future?

**Comment by heaths** [comment]
I'm not sure what steps to which you refer. Since all EXEs we run as part of the install are windows subsystem applications, batch scripts (or powershell) will not wait automatically. We do document to pass `--wait` to the bootstrapper already, but you'll need to wait on that as well.
